src/server/domain/consents/{ID}/generateChallenge.ts

```diff
@@ -0,0 +1,113 @@
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the Apache License, Version 2.0 (the 'License') and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+
+ --------------
+ ******/
+
+import { Request } from '@hapi/hapi'
+import { consentDB } from '../../../../lib/db'
+import { Consent } from '../../../../model/consent'
+import { putConsentId } from '../../../../shared/requests'
+import { promisify } from 'util'
+import { randomBytes } from 'crypto'
+import { Logger } from '@mojaloop/central-services-logger'
+const Enum = require('@mojaloop/central-services-shared').Enum
+
+/**
+ * Validates whether generate challenge request is valid
+ * by comparing consent ID sent matches with existing consent in table
+ * and if source ID matches initiator ID
+ * @param request: request received from PISP
+ * @param consent: Consent object
+ */
+export const isConsentRequestValid = function (request: Request, consent: Consent): boolean {
```

Comment:
Let's use https://github.com/hapijs/joi as the rest of Mojaloop does.

Reply:
@jgeewax just spoke with @eoln - joi doesn't seem to make sense in this situation, plus if we validate by joi it seems we'd have to add validation for all keys in the request.parameters + request.headers (based on what I just read in the joi docs) 
IIIRC your main concern was adding codes / specific error messages for different failing conditions? in that case could we implement that in the function throwing errors or returning the corresponding response code instead of the '202' code ?

Reply:
OK - and this is because the validation we're doing here isn't about the schema and format of the inputs, but instead that specific fields line up with expected environment parameters? If so, that's fine. Just want to make sure we're clear on what we're validating.

Reply:
yes, essentially we just want to check that the consent ID sent to us - correlates to a Consent object in our database AND that the initiator ID of the request matches with the initiator ID stored in the consent object.

Reply:
@jgeewax we are using `hapi-openapi` plugin which generates input validation from the OpenApi schema document: https://github.com/krakenjs/hapi-openapi#readme